core/src/main/scala/epollcat/internal/ch/SocketHelpers.scala

```diff
@@ -120,6 +136,56 @@ private[ch] object SocketHelpers {
     new InetSocketAddress(inetAddr, port)
   }
 
+  def toInet6SocketAddress(
+      addr: Ptr[posix.netinet.in.sockaddr_in6]
+  ): InetSocketAddress = {
+    val port = posix.arpa.inet.htons(addr.sin6_port).toInt
+    val addrBytes = addr.sin6_addr.at1.asInstanceOf[Ptr[Byte]]
+    val inetAddr = InetAddress.getByAddress {
+      val addr = new Array[Byte](16)
```

Comment:
As mentioned above, I will try a design study to 
simplify this after it merges.  Do not let me hold
you up (or introduce bugs you do not need).